linera-base/src/identifiers.rs

```diff
@@ -161,10 +161,12 @@ impl From<ApplicationId> for GenericApplicationId {
 
 /// A unique identifier for an application bytecode.
 #[cfg_attr(with_testing, derive(Default))]
-pub struct BytecodeId<A = ()> {
+pub struct BytecodeId<Abi = (), Parameters = (), InitializationArgument = ()> {
     /// The message ID that published the bytecode.
     pub message_id: MessageId,
-    _phantom: std::marker::PhantomData<A>,
+    _phantom_abi: std::marker::PhantomData<Abi>,
+    _phantom_parameters: std::marker::PhantomData<Parameters>,
+    _phantom_initialization_argument: std::marker::PhantomData<InitializationArgument>,
```

Comment:
We could make it a single `PhantomData<(Abi, InitializationArgument, Parameters)>`. Would save some boilerplate below.